lib/code_fund_web/controllers/api/ad_serve_controller.ex

```diff
@@ -110,12 +119,13 @@ defmodule CodeFundWeb.API.AdServeController do
       })
 
     %{
-      image: "",
-      link: "",
-      headline: "",
-      description: "",
+      image: Application.get_env(:code_fund, CodeFundWeb.Endpoint)[:default_ad_image_url] || "",
```

Comment:
In this file, i'd do like 
```
[ {:default_ad_body, default_ad_body},
 {:default_ad_headline, default_ad_headline},
 {:default_ad_image_url, default_ad_image_url},
 {:default_ad_link, default_ad_link}
] =  Application.get_env(:code_fund, AdService.Advertisement)
```
somewhere at the top of the file.
You should have variables now you can call like 
```
default_ad_link
```

Reply:
Alternatively you can call `Application.get_env/3` here instead of `/2`.
`Application.get_env(:code_fund, AdService.Advertisement, :default_ad_body)`